cmd/trace-agent/command/deprecated.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package command
+
+import (
+	"io"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cli"
+)
+
+// FixDeprecatedFlags transforms args so that they are conforming to the new CLI structure:
+// - replace non-posix flags posix flags
+// - replace deprecated flags with their command equivalents
+// - display warnings when deprecated flags are encountered
+func FixDeprecatedFlags(args []string, w io.Writer) []string {
+	var (
+		replacePid        = cli.ReplaceFlagExact("--pidfile")
+		replaceInfo       = cli.ReplaceFlagExact("info")
+		replaceVersion    = cli.ReplaceFlagExact("version")
+		replaceCpuProfile = cli.ReplaceFlagExact("--cpu-profile")
+		replaceMemProfile = cli.ReplaceFlagExact("--mem-profile")
+	)
```

Comment:
Some minor nit-picks:

Why use `var ()` instead of ` := ` notation ? Same for other files.

Also some of them are used only once and I think it's easier to read `"-cpuprofile":  cli.ReplaceFlagExact("--cpu-profile")` than `"-cpuprofile": replaceCpuProfile`

Reply:
I was just following the same pattern they had previously used in the process agent with their compatibility layer.